cmd/kubeadm/app/cmd/join_test.go

```diff
@@ -243,3 +246,233 @@ func TestNewJoinData(t *testing.T) {
 		})
 	}
 }
+
+func initJoinOptions(token string, discovery kubeadmapiv1beta1.Discovery, controlPlane bool) *joinOptions {
+	joinOptions := newJoinOptions()
+	joinOptions.token = token
+	joinOptions.externalcfg.Discovery = discovery
+	joinOptions.controlPlane = controlPlane
+	return joinOptions
+}
+
+func TestAmendExternalJoinConfiguration(t *testing.T) {
+	var tests = []struct {
+		description  string
+		args         []string
+		flags        map[string]string
+		token        string
+		controlPlane bool
+		discovery    kubeadmapiv1beta1.Discovery
+		expectedErr  bool
+		validate     func(*testing.T, *joinOptions)
+	}{
+		{
+			description: "len(opt.token) = 0",
+			args:        []string{"1.2.3.4:6443"},
```

Comment:
I don't think, that filling args here is relevant to this test case, just leave it pointing to an empty slice.
Please, avoid filling irrelevant parameters.

Reply:
if len(args) == 0, opt.externalcfg.Discovery.BootstrapToken will be set nil, and we will can't verify the value of opt.externalcfg.Discovery.BootstrapToken.Token.

Reply:
This is the "file discovery" test right? In that case we have no token and API server specified on command line, only a kubeconfig file. We expect the resulting `joinCfg.Discovery.BootstrapToken==nil` and to have `joinCfg.Discovery.File.KubeConfigPath == "https://foo"` (as specified in the `flags` field of the test case and pre-filled in `externalcfg`).